base/sparse/sparsematrix.jl

```diff
@@ -176,9 +176,9 @@ function sparse_compute_reshaped_colptr_and_rowval{Ti}(colptrS::Vector{Ti}, rowv
     if length(colptrA) != nA+1
         throw(DimensionMismatch("length of colptrA must be $(nA+1) but is $(length(colptrA))"))
     elseif maximum(colptrA) > lrowvalA+1
-        Base.throw_boundsError(A,colptrA)
+        throw(BoundsError(rowvalA, colptrA))
     elseif maxrowvalA > mA
-        Base.throw_boundsError(A,maxrowvalA)
+        throw(BoundsError(mA, maxrowvalA))
```

Comment:
I'd check what other similar sparse functions throw under this type of condition where rowval contains too-large values. This is an internal function right, is it ever called with inputs that haven't already been checked for consistency?